zipkin-lens/src/components/DependenciesPage/DependenciesGraph.jsx

```diff
@@ -0,0 +1,180 @@
+
+const useStyles = makeStyles({
+  root: {
+    // If height is 100%, scroll bar appears.
+    // I don't know that reason...
+    height: '99%',
+    zIndex: 1,
+    position: 'relative',
+  },
+  selectWrapper: {
+    position: 'absolute',
+    left: 30,
+    top: 30,
+  },
+});
+
+const reactSelectStyles = {
+  control: base => ({
+    ...base,
+    width: '15rem',
+  }),
+};
+
+const vizStyle = {
+  colorText: theme.palette.primary.contrastText,
+  colorTextDisabled: theme.palette.primary.contrastText,
+  colorConnectionLine: theme.palette.grey[800],
+  colorTraffic: {
+    normal: theme.palette.primary.dark,
+    warning: theme.palette.secondary.dark,
+    danger: theme.palette.secondary.dark,
+  },
+  colorDonutInternalColor: theme.palette.primary.light,
+  colorDonutInternalColorHighlighted: theme.palette.primary.light,
+  colorLabelBorder: theme.palette.primary.dark,
+  colorLabelText: theme.palette.primary.contrastText,
+  colorTrafficHighlighted: {
+    normal: theme.palette.primary.main,
+  },
+};
+
+const propTypes = {
+  selectedNodeName: PropTypes.string,
+  onNodeClick: PropTypes.func.isRequired,
+  edges: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+  nodes: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+};
+
+const defaultProps = {
+  selectedNodeName: '',
+};
+
+const DependenciesGraph = React.memo(({
+  selectedNodeName,
+  onNodeClick,
+  edges,
+  nodes,
+}) => {
+  const classes = useStyles();
+  const [filter, selectFilter] = useReducer((_, selected) => (selected ? selected.value : ''), '');
+
+  const filterConnections = useCallback((object, value) => {
+    if (!value) {
+      return true;
+    }
+    if (object.name === value) {
+      return true;
+    }
+    return object.source.name === value || object.target.name === value;
+  }, []);
+
+  const filterNodes = useCallback((object, value) => {
+    if (!value) {
+      return true;
+    }
+    if (object.name === value) {
+      return true;
+    }
+    return object.incomingConnections.find(conn => conn.source.name === value)
+      || object.outgoingConnections.find(conn => conn.target.name === value);
+  }, []);
+
+  const handleObjectHighlight = useCallback((highlightedObject) => {
+    if (!highlightedObject) {
+      onNodeClick(null);
+      return;
+    }
+    if (highlightedObject.type === 'node' && highlightedObject.getName() !== selectedNodeName) {
+      onNodeClick(highlightedObject.getName());
+    }
+  }, [onNodeClick, selectedNodeName]);
+
+  const maxVolume = useMemo(() => {
+    if (edges.length > 0) {
+      return edges.map(edge => edge.metrics.normal + edge.metrics.danger)
+        .reduce((a, b) => Math.max(a, b));
+    }
+    return 0;
+  }, [edges]);
+
+  const filterOptions = useMemo(() => nodes.map(node => ({
+    value: node.name,
+    label: node.name,
+  })), [nodes]);
+
+  return (
+    <Box className={classes.root}>
+      <VizceralExt
+        allowDraggingOfNodes
+        targetFramerate={30}
+        traffic={{
```

Comment:
If possible, we should set `updated` to a value that reflects when `nodes` and `edges` are actually updated to avoid `Vizceral.updateData` on every rerender. Do you think we can compute `currentTime` in the graph constructor and pass it through to here?

https://github.com/openzipkin/zipkin/blob/master/zipkin-lens/src/util/dependencies-graph.js#L15

For reference, the logic that refers to `updated` is this one

https://github.com/Netflix/vizceral-react/blob/master/src/vizceral.jsx#L130